nms/scripts/generateAPIFromSwaggerForTypeScript.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+################################################################################
+# Copyright 2022 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+set -euo pipefail
+
+USAGE="generateAPIFromSwaggerForTypeScript.sh
+— generates TypeScript NMS API bindings for swagger spec
+
+  Usage:
+    generateAPIFromSwaggerForTypeScript.sh <input> <output>
+
+  Options:
+    <input>   Input swagger.yml file to read.
+    <input>   Output file for js bindings.
+"
+
+if [ $# -ne 2 ]; then
+  echo "$USAGE"
+  exit 1
+fi
+
+INPUT=${1}
+OUTPUT=${2}
+echo "Input file: $INPUT";
+echo "Output directory: $OUTPUT";
+
+(set -x;
+yarn --silent openapi --client axios --useUnionTypes -i "${INPUT}" -o "${OUTPUT}")
+
+addHeader() {
+  local file="$1"
+  TEMPORARY_FILE=$(mktemp)
+  cat <<EOF > "${TEMPORARY_FILE}"
+/**
+ * Copyright 2022 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @generated
+ * This file is generated by "nms/scripts/generateAPIFromSwaggerForTypeScript.sh".
+ */
+$(cat "${file}")
+EOF
+  mv "${TEMPORARY_FILE}" "${file}"
+}
+
+for file in "${OUTPUT}"/*.ts "${OUTPUT}"/**/*.ts; do
```

Comment:
Are we sure all systems are already at bash 4? On mac there should still be no support for `**` in bash

Reply:
The script is executed in a docker container, through `orc8r/cloud/docker/build.py --generate` although you could in theory execute it manually.
I'm on a Mac and the script doesn't work, but only because BSD `sed -i` is incompatible with GNU `sed -i`. Which we should be able to remove after https://github.com/magma/magma/issues/12768